# Publish all joint states -----------------------------------
  joint_state_controller:
    type: "joint_state_controller/JointStateController"
    publish_rate: 500

  left_arm_controller:
    type: "position_controllers/JointTrajectoryController"
    publish_rate: 500
    joints:
      - l_arm_joint1
      - l_arm_joint2
      - l_arm_joint3
      - l_arm_joint4
      - l_arm_joint5
      - l_arm_joint6
      - l_arm_joint7
    constraints:
      goal_time: 0.0
      stopped_velocity_tolerance: 1.0

  left_hand_controller:
    type: "position_controllers/GripperActionController"
    publish_rate: 500
    joint: l_hand_joint
    action_monitor_rate: 10
    state_publish_rate:  100
    stall_velocity_threshold: 0.001
    goal_tolerance: 0.05
    stall_timeout: 0.1

  dynamixel_port:
    port_name: "/dev/sciurus17spine"
    baud_rate: 3000000
    joints:
      - l_arm_joint1
      - l_arm_joint2
      - l_arm_joint3
      - l_arm_joint4
      - l_arm_joint5
      - l_arm_joint6
      - l_arm_joint7
      - l_hand_joint
    l_arm_joint1: {id: 10, center: 2048, home: 2048, effort_const: 2.79, mode: 3  }
    l_arm_joint2: {id: 11, center: 2048, home: 3072, effort_const: 2.79, mode: 3  }
    l_arm_joint3: {id: 12, center: 2048, home: 2048, effort_const: 1.69, mode: 3  }
    l_arm_joint4: {id: 13, center: 2048, home:  271, effort_const: 1.79, mode: 3  }
    l_arm_joint5: {id: 14, center: 2048, home: 2048, effort_const: 1.79, mode: 3  }
    l_arm_joint6: {id: 15, center: 2048, home: 3413, effort_const: 1.79, mode: 3  }
    l_arm_joint7: {id: 16, center: 2048, home: 2048, effort_const: 1.79, mode: 3  }
    l_hand_joint: {id: 17, center: 2048, home: 2048, effort_const: 1.79, mode: 3  }

  joint_limits:
    l_arm_joint1:
      has_position_limits: true
      min_position: -1.5707963268
      max_position: 1.5707963268
      has_velocity_limits: true
      max_velocity: 5.969211435
      has_acceleration_limits: false
      has_jerk_limits: false
      has_effort_limits: true
      max_effort: 5.0
    l_arm_joint2:
      has_position_limits: true
      min_position: 0.0
      max_position: 1.5707963268
      has_velocity_limits: true
      max_velocity: 5.969211435
      has_acceleration_limits: false
      has_jerk_limits: false
      has_effort_limits: true
      max_effort: 5.0
    l_arm_joint3:
      has_position_limits: true
      min_position: -1.5707963268
      max_position: 1.5707963268
      has_velocity_limits: true
      max_velocity: 5.969211435
      has_acceleration_limits: false
      has_jerk_limits: false
      has_effort_limits: true
      max_effort: 5.0
    l_arm_joint4:
      has_position_limits: true
      min_position: -2.726204
      max_position: 0.0
      has_velocity_limits: true
      max_velocity: 5.969211435
      has_acceleration_limits: false
      has_jerk_limits: false
      has_effort_limits: true
      max_effort: 5.0
    l_arm_joint5:
      has_position_limits: true
      min_position: -1.5707963268
      max_position: 1.5707963268
      has_velocity_limits: true
      max_velocity: 5.969211435
      has_acceleration_limits: false
      has_jerk_limits: false
      has_effort_limits: true
      max_effort: 5.0
    l_arm_joint6:
      has_position_limits: true
      min_position: -1.047196
      max_position: 2.094395
      has_velocity_limits: true
      max_velocity: 5.969211435
      has_acceleration_limits: false
      has_jerk_limits: false
      has_effort_limits: true
      max_effort: 5.0
    l_arm_joint7:
      has_position_limits: true
      min_position: -2.967060
      max_position: 2.967060
      has_velocity_limits: true
      max_velocity: 5.969211435
      has_acceleration_limits: false
      has_jerk_limits: false
      has_effort_limits: true
      max_effort: 5.0
    l_hand_joint:
      has_position_limits: true
      min_position: -0.523598776
      max_position: 0.0
      has_velocity_limits: true
      max_velocity: 5.969211435
      has_acceleration_limits: false
      has_jerk_limits: false
      has_effort_limits: true
      max_effort: 5.0
